libr/socket/socket_http.c

```diff
@@ -71,48 +99,118 @@ static char *r_socket_http_answer (RSocket *s, int *code, int *rlen) {
 	} else {
 		res = NULL;
 	}
-fail:
+exit:
 	free (buf);
-// is 's' free'd? isn't this going to cause a double free?
+	r_buf_free (b);
 	r_socket_close (s);
 	if (rlen) {
 		*rlen = len;
 	}
 	return res;
 }
 
-R_API char *r_socket_http_get(const char *url, int *code, int *rlen) {
+#if __WINDOWS__
+static char *http_get_w32(const char *url, int *code, int *rlen) {
+	HINTERNET hInternet = InternetOpenA ("radare2 "R2_VERSION, INTERNET_OPEN_TYPE_DIRECT, NULL, NULL, 0);
+	if (!hInternet) {
+		r_sys_perror ("InternetOpenA");
+		return NULL;
+	}
+	HINTERNET hOpenUrl = InternetOpenUrlA (hInternet, url, NULL, 0, 0, 0);
+	if (!hOpenUrl) {
+		r_sys_perror ("InternetOpenUrlA");
+		InternetCloseHandle (hInternet);
+		return NULL;
+	}
+
+	char *ret = NULL;
+	size_t read_sz = 0x100000;
+	DWORD read = 0, written = 0;
```

Comment:
dont use `read` as variable names because it is already a posix symbol and makes the reading confusing, not to mention that deleting the variable definition will make crash to happen and no compile time errors, you can use a `struct {size_t r, w;} stats` and do stats.r+= read for example, or just use `r` and `w`